cmd/maya-exporter/app/collector/cstor.go

```diff
@@ -0,0 +1,211 @@
+package collector
+
+import (
+	"bytes"
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/types/v1"
+)
+
+const (
+
+	// SocketPath where istgt is listening
+	SocketPath = "/var/run/istgt_ctl_sock"
+	// HeaderPrefix is the prefix comes in the header from cstor.
+	HeaderPrefix = "iSCSI Target Controller version"
+	// EOF separates the strings from response which comes from the
+	// cstor as the collection of metrics.
+	EOF = "\r\n"
+	// Footer is used to verify if all the response has collected.
+	Footer = "OK IOSTATS"
+	// Command is a command that is used to write over wire and get
+	// the iostats from the cstor.
+	Command = "IOSTATS"
+	// BufSize is the size of response from cstor.
+	BufSize = 1024
+)
+
+// cstor implements the Exporter interface. It exposes
+// the metrics of a OpenEBS (cstor) volume.
+type cstor struct {
+	conn  net.Conn
+	stats stats
+}
+
+func Cstor() *cstor {
+	return &cstor{
+		stats: stats{},
+	}
+}
+
+// InitiateConnection tries to initiates the connection with the cstor
+// over unix domain socket. This function can not be unit tested (only
+// negative cases are possible).
+func (c *cstor) InitiateConnection(path string) error {
+	conn, err := net.Dial("unix", path)
```

Comment:
net's Dial func can be set as a field in cstor to mock & hence unit test.
e.g. 
```go
type netDialFunc func(string, string) (conn, err)

type cstor struct {
  dial netDialFunc
}
```

Reply:
Here `conn.Read` and `conn.Write` is being used per request, so i can't think of the above approach. But above approach is good for unit testing purpose. I will think about it later.

Reply:
Done